test-cases/tests/alerts/c18-validate-api-usage-alerts.md

```diff
@@ -14,148 +14,126 @@ tags:
-
-```json
-{
-  "api-usage-alert-level1": {
-    "ruleName": "RHOAMApiUsageLevel1ThresholdExceeded",
-    "level": "warning",
-    "minRate": "80%",
-    "maxRate": "90%",
-    "period": "1m"
-  },
-  "api-usage-alert-level2": {
-    "ruleName": "RHOAMApiUsageLevel2ThresholdExceeded",
-    "level": "warning",
-    "minRate": "90%",
-    "maxRate": "95%",
-    "period": "1m"
-  },
-  "api-usage-alert-level3": {
-    "ruleName": "RHOAMApiUsageLevel3ThresholdExceeded",
-    "level": "warning",
-    "minRate": "95%",
-    "period": "1m"
-  }
-}
-```
-
-4. Go to `Networking` > `Routes` under `redhat-rhmi-3scale` namespace
-5. Click on the `zync` route that starts with `https://3scale-admin...`
-6. Go to `Secrets` > `system-seed` under 3Scale namespace and copy the admin password
-7. Go back to 3Scale login page and login
-8. Click on `Ok, how does 3scale work?` and follow the 3Scale wizard to create an API
-9. Once on your API overview page, click on `Integration` on the left, then on `Configuration`
-10. Take note of the `example curl for testing` for `Staging-APIcast`
-11. Prepare the following script for each level to verify that the rate limit (100 requests/minute) works correctly:
+   ```sh
+   oc patch configmap sku-limits-managed-api-service -n redhat-rhoam-operator -p '"data": {        "rate_limit": "{\n  \"RHOAM SERVICE SKU\": {\n    \"unit\": \"minute\",\n    \"requests_per_unit\": 100\n  }\n}"    }'
+   ```
+
+4. Modify the `rate-limit-alerts` to allow alerts to fire on a per minute basis:
+
+   ```sh
+   oc patch configmap rate-limit-alerts -n redhat-rhoam-operator -p '"data": {
+       "alerts": "{\n  \"api-usage-alert-level1\": {\n    \"ruleName\": \"RHOAMApiUsageLevel1ThresholdExceeded\",\n    \"level\": \"warning\",\n    \"minRate\": \"80%\",\n    \"maxRate\": \"90%\",\n    \"period\": \"1m\"\n  },\n  \"api-usage-alert-level2\": {\n    \"ruleName\": \"RHOAMApiUsageLevel2ThresholdExceeded\",\n    \"level\": \"warning\",\n    \"minRate\": \"90%\",\n    \"maxRate\": \"95%\",\n    \"period\": \"1m\"\n  },\n  \"api-usage-alert-level3\": {\n    \"ruleName\": \"RHOAMApiUsageLevel3ThresholdExceeded\",\n    \"level\": \"warning\",\n    \"minRate\": \"95%\",\n    \"period\": \"1m\"\n  }\n}"
+   }'
+   ```
+
+5. Patch the `rhoam` CR to specify BU, SRE and Customer email addresses:
+
+   NOTE: Replace `<rh_username>` references in the below commands with a valid Red Hat username. For example: `pamccart+BU@redhat.com`
+
+   Patch BU and SRE email addresses:
+
+   ```sh
+   oc patch rhmi rhoam -n redhat-rhoam-operator --type merge -p '{"spec":{"alertingEmailAddresses":{"businessUnit":"<rh_username>+BU@redhat.com", "cssre":"<rh_username>+SRE@redhat.com"}}}'
+   ```
+
+   Patch Customer email addresses with two emails to validate multiple addresses:
+
+   ```sh
+   oc patch rhmi rhoam -n redhat-rhoam-operator --type merge -p '{"spec":{"alertingEmailAddress":"<rh_username>+CUSTOMER1@redhat.com <rh_username>+CUSTOMER2@redhat.com"}}'
```

Comment:
The format of `alertingEmailAddresses` in the cr is ```  alertingEmailAddresses:
    businessUnit: noreply-test@rhmi-redhat.com
    cssre: noreply-test@rhmi-redhat.com```
In the previous command the `businessUnit` entry is specified. Is there some key for the customer emails that should be in place? or does it sit under `alertingEmailAddress`

Reply:
it's a different key:
`alertingEmailAddress` vs `alertingEmailAddresses`

It's actually not easy to distinguish between them actually :D 

I think it would be clearer if we had a different key name for the customer email address...

Reply:
Yes I agree we should rename that one if we can, I thought it was a typo to be honest.

Reply:
Agreed, it's very confusing. I logged this ticket on the backlog yesteday: https://issues.redhat.com/browse/MGDAPI-553